server/src/main/java/org/opensearch/indices/SystemIndices.java

```diff
@@ -143,11 +141,9 @@ public boolean validateSystemIndex(String index) {
     }
 
     private static CharacterRunAutomaton buildCharacterRunAutomaton(Collection<SystemIndexDescriptor> descriptors) {
-        Optional<Automaton> automaton = descriptors.stream()
+        List<Automaton> automations = descriptors.stream()
```

Comment:
nit: I see automation in multiple place. Can we change to automaton?

Otherwise the PR lgtm.